### PR TITLE
[PLAT-1069][Hotfix] Catch Gitlab TOS Error

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Currently If a user hasn't agreed to the TOS the user is locked out of their account. This fixes that.

## Changes

- refactor `.code` to `.response_code` to catch exception better
- add descriptive messages to catch error.
screen shot 2018-08-29 at 3 46 06 pm
screen shot 2018-08-29 at 3 33 26 pm

## QA Notes

I tested this by inserting an error into the repo method like so:
```python
    def repo(self, repo_id):
        """Get a single GitLab repo's info.

        https://docs.gitlab.com/ce/api/projects.html#get-single-project

        :param str repo_id: GitLab repository id
        :return: gitlab.Project a object representing the repo
        """

        try:
            raise gitlab.GitlabGetError('You (@latifn2) must accept the Terms of Service in order to perform this action. Please access GitLab from a web browser to accept these terms.', response_code=403)
            return gitlab.Project(self.gitlab, repo_id)
        except gitlab.GitlabGetError as exc:
            if exc.response_code == 404:
                raise NotFoundError
            else:
                raise exc
```
The elusive TOS error's sentry info is here: https://sentry2.cos.io/cos/osf-back-end/issues/7576/

## Documentation

None

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-1069

### CODE REVIEW

File: addons/gitlab/models.py [hunk 1]
```diff
@@ -275,6 +276,12 @@ def before_page_load(self, node, user):
             repo = connect.repo(self.repo_id)
         except (ApiError, GitLabError):
             return
+        except gitlab.GitlabGetError as exc:
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply addons/gitlab/models.py]: 1. Yes
2. Yes

[the remark this answers is not in the document]
Comment (sloria) [reply addons/gitlab/models.py]: Ah ok. We'll probably want to handle that base error class then.

File: .docker-compose.env [hunk 2]
```diff
@@ -17,4 +17,4 @@ BROKER_URL=amqp://guest:guest@192.168.168.167:5672/
 REDIS_URL=redis://192.168.168.167:6379/1
 EMBER_DOMAIN=192.168.168.167
 
-#PYTHONUNBUFFERED=0 # This when set to 0 will allow print statements to be visible in the Docker logs
+PYTHONUNBUFFERED=0 # This when set to 0 will allow print statements to be visible in the Docker logs
```
Comment (sloria) [inline .docker-compose.env]: Why was this changed?

Comment (Johnetordoff) [reply .docker-compose.env]: Just a mistake I gotta remember to git ignore this file when debugging
